internal/gatewayapi/backendtrafficpolicy.go

```diff
@@ -482,3 +496,57 @@ func (t *Translator) buildProxyProtocol(policy *egv1a1.BackendTrafficPolicy) *ir
 
 	return pp
 }
+
+func (t *Translator) buildCircuitBreaker(policy *egv1a1.BackendTrafficPolicy) *ir.CircuitBreaker {
+	var cb *ir.CircuitBreaker
+	pcb := policy.Spec.CircuitBreaker
+
+	if pcb != nil {
+		cb = &ir.CircuitBreaker{}
+
+		if pcb.MaxConnections != nil {
+			if ui32, ok := int64ToUint32(*pcb.MaxConnections); ok {
+				cb.MaxConnections = &ui32
+			} else {
+				setCircuitBreakerPolicyErrorCondition(policy)
+				return nil
+			}
+		}
+
+		if pcb.MaxParallelRequests != nil {
+			if ui32, ok := int64ToUint32(*pcb.MaxParallelRequests); ok {
+				cb.MaxParallelRequests = &ui32
+			} else {
+				setCircuitBreakerPolicyErrorCondition(policy)
+				return nil
+			}
+		}
+
+		if pcb.MaxPendingRequests != nil {
+			if ui32, ok := int64ToUint32(*pcb.MaxPendingRequests); ok {
+				cb.MaxPendingRequests = &ui32
+			} else {
+				setCircuitBreakerPolicyErrorCondition(policy)
+				return nil
+			}
+		}
+	}
+
+	return cb
+}
+
+func setCircuitBreakerPolicyErrorCondition(policy *egv1a1.BackendTrafficPolicy) {
+	message := "Unable to translate Circuit Breaker"
```

Comment:
can we pass an extra `errMsg` to the func to add some more error msg details such as `invalid maxPendingRequests value` etc

Reply:
done